requirements-optional.txt

```diff
@@ -0,0 +1,31 @@
+# Optional dependencies of Terra that can (mostly) reliably be installed with
+# `pip`.  This file is still called `requirements-optionals.txt` just to match
+# standard pip conventions, even though none of these are required.
+#
+# If updating this, you probably want to update `qiskit.utils.optionals` too.
+
+qiskit-aer
+qiskit-toqm>=0.0.4; platform_machine != 'aarch64' or platform_system != 'Linux'
```

Comment:
I worry a bit about these 2 because of the bidirectional dependency. We've had potential issues around that in the past and the ordering here relative to the installation of qiskit-terra itself is important. It may be required to install this before or after terra in CI because of a version incompatibility and having these in the requirements list that can force it to be installed before that. I'm thinking specifically in the case of tox the order of requirements is fixed relative to the package build/install I think (or maybe there is a config option to set that I'm not sure).

Reply:
`tox` will always install these first, before the main package, but that matches what was already going on in `tox.ini`, which listed `qiskit-aer` as a dependency directly.  In Azure / GitHub, we can pass the `pip install -r requirements-optional.txt` in whatever order we like.